remind.py

```diff
@@ -0,0 +1,118 @@
+import jinja2
+from smtplib import SMTP
+from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+from functions import generate_html_file, get_osp_version, has_blockers
+
+
+def run_remind(config, blockers, server, header):
+
+	# get set of all owners in blocker file
+	owner_list = []
+	for job in blockers:
+		owners = blockers[job].get('owners', False)
+		if not owners:
+			continue
+		owner_list.extend(owners)
+	owner_set = set(owner_list)
+
+	# find each job with no blockers including the owner and send email with agg'd list
+	for owner in owner_set:
+		rows = []
+		for job_name in blockers:
+			owners = blockers[job_name].get('owners', [])
+			if owner not in owners:
+				continue
+			osp_version = get_osp_version(job_name)
+
+			# get all relevant info from jenkins
+			try:
+				job_info = server.get_job_info(job_name)
+				job_url = job_info['url']
+				lcb_num = job_info['lastCompletedBuild']['number']
+				build_info = server.get_build_info(job_name, lcb_num)
+				build_actions = build_info['actions']
+				component = [action['text'] for action in build_actions if 'COMPONENT' in action.get('text', '')]
+
+				# if build was testing specific component, find next most recent build not testing a component
+				while component != []:
+					lcb_num = lcb_num - 1
+					build_info = server.get_build_info(job_name, lcb_num)
+					build_actions = build_info['actions']
+					component = [action['text'] for action in build_actions if 'COMPONENT' in action.get('text', '')]
+
+				lcb_url = build_info['url']
+				lcb_result = build_info['result']
+				compose = [action['text'][13:-4] for action in build_actions if 'core_puddle' in action.get('text', '')]
+
+				# No compose could be found; likely a failed job where the 'core_puddle' var was never calculated
+				if compose == []:
+					compose = "Could not find compose"
+				else:
+					compose = compose[0]
+
+			except Exception as e:
+
+				# No "Last Completed Build" found
+				if job_info['builds'] == []:
+					lcb_num = None
+					compose = "N/A"
+					lcb_url = None
+					lcb_result = "NO_KNOWN_BUILDS"
+
+				# Unknown error, skip job
+				else:
+					print("Jenkins API call error on job {}: {}".format(job_name, e))
+					continue
+
+			# only care about jobs with UNSTABLE or FAILURE status that have no blockers
+			if lcb_result == "UNSTABLE" or lcb_result == "FAILURE":
+				if not has_blockers(blockers, job_name):
```

Comment:
I may have misunderstood @EliadCohen's request, but, didn't he kinda want a reminder to update blockers? :)

Reply:
I mean, sure there's a blocker, but can "remind" be used to remind the owner to examine the existing blocker and add any notes to the blocker, or update the blockers with another or bug/jira..

Reply:
I misunderstood, I guess its okay to have no reminder on stuff that has a blocker. I kinda feel, for example on a "unified" (for want of a better term) job, I may have multiple failures - lets say associated with different blockers - but only one of them is updated, the rest of the failures probably need their own blockers, but this check here doesn't care about that use case, and doesn't seek to remind me to take a look and verify if *all* blockers are set... wdyt?

Reply:
From #69:

> If the job failed and there's no blocker reported on the current blockers file, email the responsible party with a reminder to update their blocker file because so and so jobs are missing blockers.

Reply:
True, I re-read the issue, but, what about the possibility that some blockers are missed? I don't want to scope creep on you, I thought it's alright to drop line 70 and get an enhanced behavior - i.e., contact owners of failed jobs always and ask to update blockers. Do you expect someone will misuse this?

Reply:
Having now read ALL the comments, I see what you're saying and I agree there is a beneficial use case to reminding owners about all their jobs, not just untriaged. Would you be amicable to opening an RFE for such a feature so we could expand `remind()` as part of a future patch?

Reply:
Absolutely!